schema/schemaspec/schemahcl/context.go

```diff
@@ -87,10 +87,21 @@ func (s *State) setInputVals(ctx *hcl.EvalContext, body hcl.Body, input map[stri
 		}
 		ctxVars[v.Name] = v.Default
 	}
-	ctx.Variables["var"] = cty.ObjectVal(ctxVars)
+	mergeCtxVar(ctx, "var", ctxVars)
 	return nil
 }
 
+func mergeCtxVar(ctx *hcl.EvalContext, key string, vals map[string]cty.Value) {
+	v, ok := ctx.Variables[key]
+	if ok {
+		v.ForEachElement(func(key cty.Value, val cty.Value) (stop bool) {
+			vals[key.AsString()] = val
+			return false
+		})
+	}
+	ctx.Variables[key] = cty.ObjectVal(vals)
+}
```

Comment:
Why `var` is not a constant if it's used only here?
```suggestion
	mergeCtxVar(ctx, ctxVars)
	return nil
}

func mergeCtxVar(ctx *hcl.EvalContext, vals map[string]cty.Value) {
	const key = "var"
	v, ok := ctx.Variables[key]
	if ok {
		v.ForEachElement(func(key cty.Value, val cty.Value) (stop bool) {
			vals[key.AsString()] = val
			return false
		})
	}
	ctx.Variables[key] = cty.ObjectVal(vals)
}
```